src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -414,7 +414,8 @@ public static bool TreatControlCAsInput
                     throw new IOException(SR.IO_NoConsole);
 
                 int mode = 0;
-                Interop.Kernel32.GetConsoleMode(handle, out mode); // failure ignored in full framework
+                if (!Interop.Kernel32.GetConsoleMode(handle, out mode))
+                    throw Win32Marshal.GetExceptionForWin32Error(Marshal.GetLastWin32Error());
```

Comment:
It will fail when the RPC to the console server process fails. 

We do throw for SetConsoleMode call later that does the same RPC and it is exposed to the same failure modes.

Ignoring the error here can lead to corrupting the console mode bits (we will reset all of them) if the error is intermittent

Reply:
Ok, thanks.